Lib/logging/__init__.py

```diff
@@ -1828,6 +1828,25 @@ def getChild(self, suffix):
             suffix = '.'.join((self.name, suffix))
         return self.manager.getLogger(suffix)
 
+    def getChildren(self):
+
+        def _hierlevel(logger):
+            if logger is logger.manager.root:
+                return 0
+            return 1 + logger.name.count('.')
+
+        d = self.manager.loggerDict
+        _acquireLock()
+        try:
+            # exclude PlaceHolders - the last check is to ensure that lower-level
+            # descendants aren't returned - if there are placeholders, a logger's
+            # parent field might point to a grandparent or ancestor thereof.
+            return set([item for item in d.values() \
+                if isinstance(item, Logger) and item.parent is self and \
+                    _hierlevel(item) == 1 + _hierlevel(item.parent)])
```

Comment:
Building a list can be avoided!  (+ style details)

```suggestion
            return set(item for item in d.values()
                       if isinstance(item, Logger) and item.parent is self and
                       _hierlevel(item) == 1 + _hierlevel(item.parent))
```

Reply:
Oh, yes, good point!